mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -586,65 +600,67 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
                 )
                 break;
             }
-
-            // Only create a new cell if the AI generated code in its response
-            if (aiMessage?.message) {
-                const aiGeneratedCode = getCodeBlockFromMessage(aiMessage.message);
-                if (aiGeneratedCode) {
-                    // Insert a new cell for the next step
-                    await app.commands.execute("notebook:insert-cell-below")
-
-                    // Wait for the new cell to be created
-                    await sleep(1000)
-                }
-            }
         }
 
         setAgentExecutionStatus('idle')
     }
 
-    const updateCodeDiffStripes = (aiMessage: OpenAI.ChatCompletionMessageParam | undefined) => {
+    const updateCodeDiffStripes = (aiMessage: OpenAI.ChatCompletionMessageParam | undefined, updateCellID: string) => {
         if (!aiMessage) {
             return
         }
 
-        const codeCellID = getActiveCellID(notebookTracker)
-        const activeCellCode = getCellCodeByID(notebookTracker, codeCellID)
+        const updateCellCode = getCellCodeByID(notebookTracker, updateCellID)
 
-        if (codeCellID === undefined || activeCellCode === undefined) {
+        if (updateCellID === undefined || updateCellCode === undefined) {
             return
         }
 
         // Extract the code from the AI's message and then calculate the code diffs
         const aiGeneratedCode = getCodeBlockFromMessage(aiMessage);
         const aiGeneratedCodeCleaned = removeMarkdownCodeFormatting(aiGeneratedCode || '');
-        const { unifiedCodeString, unifiedDiffs } = getCodeDiffsAndUnifiedCodeString(activeCellCode, aiGeneratedCodeCleaned)
+        const { unifiedCodeString, unifiedDiffs } = getCodeDiffsAndUnifiedCodeString(updateCellCode, aiGeneratedCodeCleaned)
 
 
         // Store the code cell ID where we write the code diffs so that we can
         // accept or reject the code diffs to the correct cell
-        cellStateBeforeDiff.current = { codeCellID: codeCellID, code: activeCellCode }
+        cellStateBeforeDiff.current = { codeCellID: updateCellID, code: updateCellCode }
 
         // Temporarily write the unified code string to the active cell so we can display
         // the code diffs to the user
-        writeCodeToCellByID(notebookTracker, unifiedCodeString, codeCellID)
+        writeCodeToCellByID(notebookTracker, unifiedCodeString, updateCellID)
         updateCodeCellsExtensions(unifiedDiffs)
 
         // Briefly highlight the code cell to draw the user's attention to it
-        highlightCodeCell(notebookTracker, codeCellID)
+        highlightCodeCell(notebookTracker, updateCellID)
     }
 
     const displayOptimizedChatHistory = chatHistoryManager.getDisplayOptimizedHistory()
 
-    const previewAICode = () => {
+    const previewAICodeToCellID = (cellID: string) => {
```

Comment:
If we don't use this anywhere else, let's put this logic back in previewAICodeToActiveCell